kolibri/content/api.py

```diff
@@ -281,6 +283,43 @@ def all_content(self, request, **kwargs):
         return Response(serializer.data)
 
 
+class ContentNodeGranularViewset(mixins.RetrieveModelMixin, viewsets.GenericViewSet):
+    serializer_class = serializers.ContentNodeGranularSerializer
+
+    def get_queryset(self, available=False):
```

Comment:
This can probably be optimized by doing some prefetches. We can merge as is, but should look at doing some prefetch_related calls in order to optimize the child fetches.

Reply:
Sorry for not noticing this previously, but because `available` is a boolean flag, it might be better to set it to `None` when you don't want to query by it.